Tests/RegexTests/ParseTests.swift

```diff
@@ -2711,6 +2720,26 @@ extension RegexTests {
       """, .cannotRemoveExtendedSyntaxInMultilineMode
     )
 
+    diagnosticWithDelimitersTest(#"""
+      #/
+      \Q
+      \E
+      /#
+      """#, .quoteMayNotSpanMultipleLines)
+
+    diagnosticWithDelimitersTest(#"""
+      #/
+        \Qabc
+          \E
+      /#
+      """#, .quoteMayNotSpanMultipleLines)
+
+    diagnosticWithDelimitersTest(#"""
+      #/
+        \Q
+      /#
+      """#, .quoteMayNotSpanMultipleLines)
```

Comment:
Should this just be the regex `/\Q/`, i.e. it's not spanning multiple lines because there's no additional line? I'm ok either way, and perhaps reporting an error is helpful.

Reply:
Good point! Currently we don't do any whitespace stripping, so if we just allowed it as-is, it would parse as `\Q\n       \E`. That being said, it seems reasonable enough to strip everything after-and-including the last newline given we enforce that the closing delimiter must appear on a newline. Does that sound reasonable to you, or should be leave it as an error for now?

Reply:
I could go either way, what do you think?

Reply:
I think we should probably have a consistent behavior between `(?-x)` and unterminated `\Q` whichever way we go. For now at least, I'm tempted to say we should keep everything using the rule where semantic whitespace cannot span multiple lines anywhere in the literal, so `(?-x)` and unterminated `\Q` are always invalid. So this test case would remain as-is. Given this rule only affects literals, we'd be free to lift in the future if we wanted. Does that sound reasonable?

Reply:
SGTM